src/templates/homepage/HeroSection/HeroSection.tsx

```diff
@@ -0,0 +1,158 @@
+            <p
+              className={getClassNames(editorStyles, [
+                "has-text-weight-semibold",
+                "has-text-white",
+                "key-highlight-title",
+                "is-uppercase",
+                "padding--top--xs",
+              ])}
+            >
+              {title}
+            </p>
+          ) : null}
+          {/* Description */}
+          {description ? (
+            <p
+              className={getClassNames(editorStyles, [
+                "has-text-white-trans",
+                "padding--bottom--sm",
+              ])}
+            >
+              {description}
+            </p>
+          ) : null}
+        </div>
+      </div>
+    ) : null}
+  </>
+)
+interface KeyHighlightsProps {
+  highlights: KeyHighlightElemProps[]
+}
+const KeyHighlights = ({ highlights }: KeyHighlightsProps) => (
+  <section
+    id={editorStyles["key-highlights"]}
+    className={getClassNames(editorStyles, ["bp-section", "is-paddingless"])}
+  >
+    <div
+      id="key-highlights"
+      className={getClassNames(editorStyles, ["bp-container"])}
+    >
+      <div
+        className={getClassNames(editorStyles, [
+          "row",
+          "is-gapless",
+          "has-text-centered",
+        ])}
+      >
+        {highlights.map(({ title, description }, idx) => (
+          <KeyHighlightElem
+            title={title}
+            description={description}
+            key={`${title}-${idx}`}
+          />
+        ))}
+      </div>
+    </div>
+  </section>
+)
+
+interface TemplateHeroSectionProps {
+  hero: {
+    background?: string
+    title?: string
+    subtitle?: string
+    button?: string
+    dropdown: {
+      title: string
+      options: { title: string }[]
+    }
+    // eslint-disable-next-line camelcase
+    key_highlights: {
+      title?: string
+      description?: string
+    }[]
+  }
+  dropdownIsActive: boolean
+  toggleDropdown: () => void
+  variant: HeroBannerLayouts
+}
+
+export const TemplateHeroSection = forwardRef<
+  HTMLDivElement,
+  TemplateHeroSectionProps
+>(
+  (
+    {
+      hero,
+      dropdownIsActive,
+      toggleDropdown,
+      variant,
+    }: TemplateHeroSectionProps,
+    ref
+  ) => {
+    const loadedImageURL = useFetchPreviewMedia(hero.background)
+
+    const heroStyle = {
+      // See j08691's answer at https://stackoverflow.com/questions/21388712/background-size-doesnt-work
+      background: `url(${loadedImageURL}) no-repeat center center/cover`,
+    }
+    return (
+      <div ref={ref}>
+        {/* Main hero banner */}
+        <section
+          className={getClassNames(editorStyles, ["bp-hero", "bg-hero"])}
+          style={heroStyle}
+        >
+          {variant === "center" ? (
+            <HeroCenteredLayout
+              hero={hero}
+              dropdownIsActive={dropdownIsActive}
+              toggleDropdown={toggleDropdown}
+            />
+          ) : (
+            <HeroImageOnlyLayout />
+          )}
```

Comment:
Do we have a fallback for when variant doesn't exist?

Reply:
i'll raise a PR at the end so that i can account for all the variants!